### DISCUSSION TITLE
Static URLs calling links and assets

### DISCUSSION THREAD

**Discussion by mrburns-ch** [kromitgmbh/titra#84] [Ideas]
Dear kromitgmbh

Thanks for this great working docker image of Titra!

What I realized is, that there are a lot of static urls in the code.
Usually this isn't really an issue. But in some cases, for example, when your baseurl (example: behind an nginx proxy) is "https://servername:8443/titra" it can cause troubles. 

It tries to load https://servername:8443/favicons
or the registerbuttons calls 
https://servername:8443/join instead of

https://servername:8443/titra/favicons 
or
https://servername:8443/titra/join

Could this be solved from your side somehow? I already tried a lot with rewrite rules in proxy but it seems endless. I also see conflicts with future configs.

I'm working in a complex environment where dns requests and ssl certificates can take a lot of time. So we tried to implement it with this "location" proxy. It can happen to others as well :-).

Best Regards

**Comment by faburem** [comment]
Hi,
thanks for the feedback! In general the URLs are not hardcoded but we might make too many assumptions about the server (reverse proxy) configuration. The assets and links are using the relative path to the root URL e.g "/favicons". After researching it a bit I am not sure this can be handled on the application side though (happy if you proof me wrong 🤓). Have you tried the proxy_pass option in your nginx proxy as suggested in this thread: https://serverfault.com/questions/932628/how-to-handle-relative-urls-correctly-with-a-nginx-reverse-proxy ?

Best,
Fabian
